apm-sniffer/bootstrap-plugins/jdk-threading-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdk/threading/ThreadingMethodInterceptor.java

```diff
@@ -34,11 +34,10 @@
     public void beforeMethod(final EnhancedInstance objInst, final Method method, final Object[] allArguments,
         final Class<?>[] argumentsTypes, final MethodInterceptResult result) {
 
-        AbstractSpan span = ContextManager.createLocalSpan(generateOperationName(objInst, method));
-        span.setComponent(ComponentsDefine.JDK_THREADING);
-
         final Object storedField = objInst.getSkyWalkingDynamicField();
         if (storedField != null) {
+            AbstractSpan span = ContextManager.createLocalSpan(generateOperationName(objInst, method));
+            span.setComponent(ComponentsDefine.JDK_THREADING);
```

Comment:
I am not sure this is the right fix. Why not remove `afterMethod`'s `ContextManager.stopSpan()`, as we are always creating local span in `beforeMethod`?

Reply:
Yes, `beforeMethod` is always creating local span，and `afterMethod` can not clear `ContextManager`'s `CONTEXT` and `RUNTIME_CONTEXT`, that will cause a memory leak.

Add `objInst.getSkyWalkingDynamicField()` check，the problem of memory leak can be solved by keeping the logic in beforeMethod.

In the ThreadingConstructorInterceptor class:
```java
       // in our sence ContextManager.isActive() will return false，cause 
        if (ContextManager.isActive()) {
            objInst.setSkyWalkingDynamicField(ContextManager.capture());
        }
```
In the ThreadingMethodInterceptor  class:
```
    @Override
    public void beforeMethod(final EnhancedInstance objInst, final Method method, final Object[] allArguments,
        final Class<?>[] argumentsTypes, final MethodInterceptResult result) {

        // LocalSpan is always created here and referenced by the worker of ThreadPool
        AbstractSpan span = ContextManager.createLocalSpan(generateOperationName(objInst, method));
        span.setComponent(ComponentsDefine.JDK_THREADING);

        final Object storedField = objInst.getSkyWalkingDynamicField();
        if (storedField != null) {
            final ContextSnapshot contextSnapshot = (ContextSnapshot) storedField;
            ContextManager.continued(contextSnapshot);
        }

    }

    @Override
    public Object afterMethod(final EnhancedInstance objInst, final Method method, final Object[] allArguments,
        final Class<?>[] argumentsTypes, final Object ret) {
        // ContextManager.CONTEXT cannot be cleaned up here
        final Object storedField = objInst.getSkyWalkingDynamicField();
        if (storedField != null) {
            ContextManager.stopSpan();
        }

        return ret;
    }
`````

Reply:
Do you really read what I said? I never said there is no memory leak. I was saying what is the better way to fix.

Reply:
OK, I'll recheck if there's something wrong with our known scenes, then submit a new version.

Reply:
The key is, if a `runnable#run` is called, when should create a local span. Your logic is, only creating when dynamicField has value, my question is should we create it always, just with or without reference.

Reply:
`beforeMethod` should always create local span，otherwise the trace of some scenarios will be missing. Just make sure that `afterMethod` calls the `ContextManager.stopSpan()` method to clear the `ContextManager`'s `CONTEXT` and `RUNTIME_CONTEXT` reference, it won't cause a memory leak.

Reply:
What are you doing? You are still don't using the same condition to create and stop span. I am very confused.

Reply:
I'm sorry there was a problem with the code submission.